documentation/meta/maintenance/elasticsearch_cluster.md

```diff
@@ -0,0 +1,349 @@
+`elasticsearch-8.9.0`[^past-foibles]. Make changes to the configuration based on
+the release note reading done in previous steps (especially step 0), and deploy
+the new cluster. It may take several attempts to successfully deploy the new
+cluster. The critical things to ensure are:
+
+1. All nodes successfully register with the cluster, including the expected
+   number of data and master nodes based on the configuration.
+1. All nodes have the correct resources allocated to the Elasticsearch process.
+1. All configuration changes are reflected in the new cluster.
+
+[Elasticvue is especially useful for this](https://elasticvue.com/), in
+particular the "Nodes" tab, which gives a good overview of the cluster's nodes.
+It also makes it easy to send REST API calls to the cluster for confirming
+configurations if needed.
+
+Tearing down and redeploying the new cluster over and over for each
+configuration change during the initial deployment attempts can be cumbersome
+and slow. Sometimes it will be easier to follow the
+[instructions for in place configuration changes](#in-place-changes-to-live-nodes)
+on the nodes in the new cluster until we have determined the correct final
+configuration. The final successful deployment _must_ be from scratch: i.e., do
+not treat an in-place configuration change as a success until the configuration
+is verified to work for a wholly new cluster. Clusters must be able to spin up
+in a fully configured state without further intervention for the module to be
+considered usable.
+
+[^past-foibles]:
+    In the past we reused the single module and made changes in place. This
+    happened to work out, but if something catastrophic had happened, and we
+    needed to deploy the current cluster from scratch again, we would have had
+    our work cut out for us, juggling reverts and likely Terraform state issues.
+    This comes down to the fact that our Elasticsearch module is inflexible,
+    with configuration created inside the module, rather than defined at the
+    root module level. Because it's simpler to duplicate the module rather than
+    add conditionals inside the module to handle multiple versions, the
+    recommendation is to duplicate it. This ensures that we can redeploy the
+    current cluster, whether entirely or just a failed node, without needing to
+    juggle changes to the code configuring that cluster.
+
+Deploy a staging cluster first. Production will come later. The cluster boxes
+are expensive, so minimising the length of time that we have multiple cluster
+versions deployed per environment helps save us some money and makes the list of
+EC2 boxes a little less confusing to navigate.
+
+```{warning}
+The existing clusters will be deprovisioned at the end of the process and
+should be left alone in the meantime. They are still being used by staging
+and production until the final steps!
+```
+
+#### 5. Run a reindex with the new staging cluster
```

Comment:
I wonder how restoring a backup between versions, if possible, compares in cost and time to a total reindexing? 💭

Reply:
It might be possible, but upgrading between major versions doesn't allow you to do that, at least not 7 to 8. The data nodes use a separate block device, the SSD, that would be easy to image and restore from, but I don't know what benefit that gives us. We still need to test indexing against the new cluster anyway, so it's not like we could skip reindexing on staging and a data refresh.